infra/es_cluster.tf

```diff
@@ -151,12 +156,26 @@ EOF
 docker build -t es .
 docker run -d \
            --name es \
-           -p 80:9200 \
+           -p 9200:9200 \
            -p 9300:9300 \
            -e ES_JAVA_OPTS="-Xmx6g -Xms6g" \
            es
 
-docker logs -f es
+docker run -d \
+           --name kibana \
+           -p 5601:5601 \
+           --link es:elasticsearch \
+           -e TELEMETRY_ENABLED=false \
+           docker.elastic.co/kibana/kibana:7.13.2
+
+## Getting container output directly to the GCP console
+
+( exec 1> >(while IFS= read -r line; do echo "elastic: $line"; done); docker logs -f es ) &
```

Comment:
maybe overkill but do they have an actual healthcheck that we can ping instead? Or even better, maybe they use the proper [docker healthchecks](https://docs.docker.com/engine/reference/builder/#healthcheck). Definitely not a blocker if it’s not easy to use either of those.

Reply:
Not quite sure why you comment here, this is just log redirection.

In terms of healthcheck, what we want to know at the load balancer/instance manager level is whether or not there is a process listening for HTTP connections. Checking that the port is open seems good enough for that; the actual health endpoints on ES give information about the state of the cluster as a whole, which is completely irrelevant for load balancing and potentially damaging for "kill node" decisions the instance manager would make.

On the Kibana side, the status endpoint seems to report on two things: whether it can connect to ES (given that this is an in-machine connection, the only way this doesn't work is if the ES container is down, in which case the machine will be killed by the ES healthcheck) and whether all plugins are correctly installed with their dependencies etc., which is unlikely to change.

Finally, regarding Docker health check, I don't think there's much value in having Docker try to restart the container vs. just starting a new machine.

It was worth looking into, so thanks for prompting me, but I think I'd rather leave the setup as it is.

Reply:
oh I somehow misread this as waiting until we see a specific line in the logs which it clearly doesn’t do. need more coffee.

Reply:
Oh, no; I just want the logs of both containers to appear as logs emitted by the init script, so they show up in the "serial output" screen in the GCP console. And while I'm at it I prefix them with either `elastic:` or `kibana:` so I know which is which.